functional/FunctionalTest.java

```diff
@@ -3795,6 +3798,155 @@ public static void getDefaultRetention_test() throws Exception {
+        }
+
+        ZonedDateTime retentionUntil = ZonedDateTime.now(Time.UTC).plusDays(3).withNano(0);
+        Retention expectedConfig = new Retention(RetentionMode.GOVERNANCE, retentionUntil);
+        client.setObjectRetention(
+            SetObjectRetentionArgs.builder()
+                .bucket(bucketName)
+                .object(objectName)
+                .config(expectedConfig)
+                .build());
+
+        Retention config =
+            client.getObjectRetention(
+                GetObjectRetentionArgs.builder().bucket(bucketName).object(objectName).build());
+
+        if (!(config.retainUntilDate().equals(expectedConfig.retainUntilDate()))
+            || (config.mode() != expectedConfig.mode())) {
+          throw new Exception(
+              "[FAILED] Expected: expected duration : "
+                  + expectedConfig.retainUntilDate()
+                  + ", got: "
+                  + config.retainUntilDate()
+                  + " expected mode :"
+                  + expectedConfig.mode()
+                  + ", got: "
+                  + config.mode());
+        }
+
+        // You can perform operations on object versions that are locked in governance mode as if
+        // they were unprotected if you have the s3:bypassGovernanceMode permission. These
+        // operations include deleting an object version, shortening the retention period, or
+        // removing the Object Lock by placing a new lock with empty parameters.
+        ZonedDateTime shortenedRetentionUntil = ZonedDateTime.now(Time.UTC).plusDays(1).withNano(0);
+        Retention expectedConfigWithShortenedPeriod =
+            new Retention(RetentionMode.GOVERNANCE, shortenedRetentionUntil);
+        client.setObjectRetention(
+            SetObjectRetentionArgs.builder()
+                .bucket(bucketName)
+                .object(objectName)
+                .config(expectedConfigWithShortenedPeriod)
+                .bypassGovernanceMode(true)
+                .build());
+
+        Retention updatedConfig =
+            client.getObjectRetention(
+                GetObjectRetentionArgs.builder().bucket(bucketName).object(objectName).build());
+
+        if (!(updatedConfig
+                .retainUntilDate()
+                .equals(expectedConfigWithShortenedPeriod.retainUntilDate()))
+            || (updatedConfig.mode() != expectedConfigWithShortenedPeriod.mode())) {
+          throw new Exception(
+              "[FAILED] Expected: expected duration : "
+                  + expectedConfigWithShortenedPeriod.retainUntilDate()
+                  + ", got: "
+                  + updatedConfig.retainUntilDate()
+                  + " expected mode :"
+                  + expectedConfigWithShortenedPeriod.mode()
+                  + ", got: "
+                  + updatedConfig.mode());
+        }
+
+        // Set empty object lock configuration by setting bypassGovernanceMode as true
```

Comment:
You could fix `object lock configuration` comment overall. This comment will make confusion with `{Get,Put}ObjectLockConfiguration`